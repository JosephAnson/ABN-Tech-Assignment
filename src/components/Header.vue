<script setup lang="ts">
const route = useRoute()

const name = ref(
  route && "name" in route.params
    ? getRouterParamsAsString(route.params.name)
    : ""
)
const router = useRouter()

const search = (name: string) => {
  if (name) {
    router.push(`/find/${name}`)
  }
}
</script>

<template>
  <div class="relative bg-primary-100">
    <div class="container">
      <div class="flex items-center justify-between py-2">
        <div
          class="heading font-secondary font-semibold flex justify-start text-l mr-2"
        >
          <router-link to="/"> TV SHOW </router-link>
        </div>
        <div class="items-center justify-end flex flex-1">
          <Search v-model:value="name" @enter="search(name)" />
        </div>
      </div>
    </div>
  </div>
</template>
