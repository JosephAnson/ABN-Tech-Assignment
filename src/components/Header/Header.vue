<script setup lang="ts">
import Search from "~/components/Search/Search.vue"

const route = useRoute()

const name = $ref(route.params.name)
const router = useRouter()

const search = () => {
  if (name) {
    router.push(`/find/${name}`)
  }
}
</script>

<template>
  <div class="relative bg-primary-100">
    <div class="container">
      <div class="flex items-center justify-between py-2">
        <div
          class="heading font-secondary font-semibold flex justify-start text-l mr-2"
        >
          <router-link to="/"> TV SHOW </router-link>
        </div>
        <div class="items-center justify-end flex flex-1">
          <Search v-model:value="name" @enter="search" />
        </div>
      </div>
    </div>
  </div>
</template>
