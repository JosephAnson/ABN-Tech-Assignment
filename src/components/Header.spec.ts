import userEvent from "@testing-library/user-event"
import { cleanup, render } from "@testing-library/vue"
import { afterEach, describe, expect, test, vi } from "vitest"
import { useRoute, useRouter } from "vue-router"
import Header from "./Header.vue"

vi.mock("vue-router")

function createRenderHeader(name = "DefaultName") {
  vi.mocked(useRoute, { partial: true }).mockReturnValue({
    params: {
      name,
    },
  })

  vi.mocked(useRouter, { partial: true }).mockReturnValue({
    push: vi.fn().mockImplementation((name) => name),
  })

  return render(Header, {
    global: {
      components: {
        RouterLink: {
          template: `<a><slot /></a>`,
        },
      },
    },
  })
}

describe("Header", () => {
  afterEach(() => cleanup())

  test("should render header", () => {
    const { html } = createRenderHeader()
    expect(html()).toMatchSnapshot()
  })

  test("should not search title", async () => {
    const { getByRole } = createRenderHeader("")
    await userEvent.click(getByRole(/button/i))

    expect(useRouter().push).not.toHaveBeenCalled()
  })

  test("should search title", async () => {
    const { getByRole } = createRenderHeader("")

    const name = "Test Search 1"

    const textBox = getByRole(/searchbox/i)
    await userEvent.type(textBox, name)
    await userEvent.click(getByRole(/button/i))

    expect(useRouter().push).toHaveBeenCalledWith(`/find/${name}`)
  })
})
