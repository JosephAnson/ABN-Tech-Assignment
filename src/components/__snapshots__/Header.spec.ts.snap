// Vitest Snapshot v1

exports[`Header > should render header 1`] = `
"<div class=\\"relative bg-primary-100\\">
  <div class=\\"container\\">
    <div class=\\"flex items-center justify-between py-2\\">
      <div class=\\"heading font-secondary font-semibold flex justify-start text-l mr-2\\"><a to=\\"/\\"> TV SHOW </a></div>
      <div class=\\"items-center justify-end flex flex-1\\">
        <div class=\\"flex items-center relative shadow-sm bg-gray-800 sm:text-sm border-gray-300 rounded-md\\"><input type=\\"search\\" class=\\"bg-transparent w-full px-4 block h-8 focus:ring-indigo-500 focus:border-indigo-500\\" placeholder=\\"Search TV shows\\" autocomplete=\\"off\\"><button class=\\"flex w-10 h-8 p-1 cursor-pointer z-10 hover:bg-gray-700 rounded-r-md\\"><span class=\\"i-carbon-search w-full h-full\\"></span><span class=\\"sr-only\\">Search Button</span></button></div>
      </div>
    </div>
  </div>
</div>"
`;
